### PR TITLE
Linear limiter

### PR DESCRIPTION
**Pull request by direvius**

### CODE REVIEW

File: limiter/linear.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+loop:
+	for n := 0.0; n < maxCount; n += 1.0 {
+		ts, err := quadraticRightRoot(a, b, -n)
+		if err != nil {
+			return err
+		}
+		waitPeriod := ts - time.Since(startTime).Seconds()
+		if waitPeriod > 0 {
+			select {
+			case <-time.After(time.Duration(waitPeriod*1e9) * time.Nanosecond):
```
Comment (m0sth8) [inline limiter/linear.go]: it's better to use time.Second instead of 1e9

File: limiter/linear.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+				break loop
+
+			}
+		}
+	}
+	// now wait until the end of specified period
+	waitPeriod := l.period.Seconds() - time.Since(startTime).Seconds()
+	if waitPeriod > 0 {
+		select {
+		case <-time.After(time.Duration(waitPeriod*1e9) * time.Nanosecond):
```
Comment (m0sth8) [inline limiter/linear.go]: it's better to use time.Second instead of 1e9

File: limiter/linear.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	return nil
+}
+
+func NewLinear(startRps, endRps, period float64) (l Limiter) {
+	return &linear{
+		// timer-based limiters should have big enough cache
+		limiter:  limiter{make(chan struct{}, 65536)},
+		startRps: startRps,
+		endRps:   endRps,
+		period:   time.Duration(period*1e9) * time.Nanosecond,
```
Comment (m0sth8) [inline limiter/linear.go]: it's better to use time.Second instead of 1e9

Comment (direvius) [inline limiter/linear.go]: need less then second precision here

**Pull request merged**